retention-autocreate = false : crash !

Hello,

When `retention-autocreate = false`, if you do not create a retention policy, InfluxDB will crash.
I do not expect that it works in that condition, but at least it would put something explicit in the log and either stop gracefully or ignore the metrics. But not crash !

Running on Redhat 6 with influxdb-0.9.5_nightly_afe5cc6-1.x86_64

Here is how to reproduce.

## 1/ Start from scratch
```
# rm -rf /var/opt/influxdb/meta /var/opt/influxdb/hh /var/opt/influxdb/data /var/opt/influxdb/wal /var/log/influxdb/influxd.log
# /etc/init.d/influxdb start
```

## 2/ notice the configuration changes :

```
# diff /etc/opt/influxdb/influxdb.conf.rpmnew /etc/opt/influxdb/influxdb.conf
[...]
22c22
<   retention-autocreate = true
---
>   retention-autocreate = false
202,204c202,204
<   enabled = false
<   # bind-address = ""
<   # database = ""
---
>   enabled = true
>   bind-address = ":4242"
>   database = "test1"
```
Notice the very important `retention-autocreate = false`.

## 3/ Check that all is going well...
```
# curl -s -G 'http://localhost:8086/query' --data-urlencode 'q=SHOW DATABASES' | jq '.'
{
  "results": [
    {
      "series": [
        {
          "values": [
            [
              "test1"
            ],
            [
              "_internal"
            ]
          ],
          "columns": [
            "name"
          ],
          "name": "databases"
        }
      ]
    }
  ]
}
```

```
# curl -s -G 'http://localhost:8086/query' --data-urlencode 'q=SHOW RETENTION POLICIES ON test1' | jq '.'
{
  "results": [
    {
      "series": [
        {
          "columns": [
            "name",
            "duration",
            "replicaN",
            "default"
          ]
        }
      ]
    }
  ]
}
```

Note : jq rocks !!!

## 4/ Crash Influxdb
Insert a point in the database (using the opentsdb plugin - not tested with other plugins)
```
# echo "put metric "$(date +"%s")" 1 mytag=TAG" | nc localhost 4242
```

Enjoy !

Here is the full log below (except the 2 lines starting with `[http]` containing the 2 queries above).

```
2015/09/16 17:02:07 InfluxDB starting, version 0.9.5-nightly-afe5cc6, branch master, commit afe5cc67f40e3a245182b983934b2105b7a75314
2015/09/16 17:02:07 Go version go1.5, GOMAXPROCS set to 4
2015/09/16 17:02:07 Using configuration at: /etc/opt/influxdb/influxdb.conf
[metastore] 2015/09/16 17:02:07 Using data dir: /var/opt/influxdb/meta
[metastore] 2015/09/16 17:02:07 Node at HOST.DOMAIN:8088 [Follower]
[metastore] 2015/09/16 17:02:09 Node at HOST.DOMAIN:8088 [Leader]. peers=[]
[metastore] 2015/09/16 17:02:09 Created local node: id=1, host=HOST.DOMAIN:8088
[monitor] 2015/09/16 17:02:09 Starting monitor system
[monitor] 2015/09/16 17:02:09 'build' registered for diagnostics monitoring
[monitor] 2015/09/16 17:02:09 'runtime' registered for diagnostics monitoring
[monitor] 2015/09/16 17:02:09 'network' registered for diagnostics monitoring
[monitor] 2015/09/16 17:02:09 'system' registered for diagnostics monitoring
[store] 2015/09/16 17:02:09 Using data dir: /var/opt/influxdb/data
[handoff] 2015/09/16 17:02:09 Starting hinted handoff service
[handoff] 2015/09/16 17:02:09 Using data dir: /var/opt/influxdb/hh
[tcp] 2015/09/16 17:02:09 Starting cluster service
[shard-precreation] 2015/09/16 17:02:09 Starting precreation service with check interval of 10m0s, advance period of 30m0s
[snapshot] 2015/09/16 17:02:09 Starting snapshot service
[copier] 2015/09/16 17:02:09 Starting copier service
[admin] 2015/09/16 17:02:09 Starting admin service
[admin] 2015/09/16 17:02:09 Listening on HTTP: 0.0.0.0:8083
[continuous_querier] 2015/09/16 17:02:09 Starting continuous query service
[httpd] 2015/09/16 17:02:09 Starting HTTP service
[httpd] 2015/09/16 17:02:09 Authentication enabled: false
[httpd] 2015/09/16 17:02:09 Listening on HTTP: 0.0.0.0:8086
[opentsdb] 2015/09/16 17:02:09 Starting OpenTSDB service
[monitor] 2015/09/16 17:02:09 Storing statistics in database '_internal' retention policy '', at interval 10s
[metastore] 2015/09/16 17:02:09 database '_internal' created
[metastore] 2015/09/16 17:02:09 database 'test1' created
[opentsdb] 2015/09/16 17:02:09 Listening on: 0.0.0.0:4242
[retention] 2015/09/16 17:02:09 Starting retention policy enforcement service with check interval of 30m0s
[run] 2015/09/16 17:02:09 Listening for signals
[metastore] 2015/09/16 17:02:09 retention policy 'monitor' for database '_internal' created
[query] 2015/09/16 17:02:16 SHOW DATABASES
[wal] 2015/09/16 17:02:19 WAL starting with 30720 ready series size, 0.50 compaction threshold, and 20971520 partition size threshold
[wal] 2015/09/16 17:02:19 WAL writing to /var/opt/influxdb/wal/_internal/monitor/1
[query] 2015/09/16 17:02:21 SHOW RETENTION POLICIES test1
[wal] 2015/09/16 17:02:25 Flush due to idle. Flushing 2 series with 2 points and 178 bytes from partition 1
[wal] 2015/09/16 17:02:25 write to index of partition 1 took 5.511582ms
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x20 pc=0x630733]

goroutine 38 [running]:
github.com/influxdb/influxdb/cluster.(*PointsWriter).MapShards(0xc820114e80, 0xc820198980, 0xc8201d03d0, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cluster/points_writer.go:179 +0x1f3
github.com/influxdb/influxdb/cluster.(*PointsWriter).WritePoints(0xc820114e80, 0xc820198980, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cluster/points_writer.go:215 +0x1d9
github.com/influxdb/influxdb/services/opentsdb.(*Service).processBatches(0xc820128460, 0xc820119b00)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/opentsdb/service.go:349 +0x278
created by github.com/influxdb/influxdb/services/opentsdb.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/opentsdb/service.go:129 +0x9c3

goroutine 1 [chan receive]:
main.(*Main).Run(0xc8200f9f00, 0xc82000a060, 0x4, 0x4, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cmd/influxd/main.go:91 +0xa16
main.main()
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cmd/influxd/main.go:42 +0x29b

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /root/.gvm/gos/go1.5/src/runtime/asm_amd64.s:1696 +0x1

goroutine 5 [syscall]:
os/signal.loop()
        /root/.gvm/gos/go1.5/src/os/signal/signal_unix.go:22 +0x18
created by os/signal.init.1
        /root/.gvm/gos/go1.5/src/os/signal/signal_unix.go:28 +0x37

goroutine 19 [IO wait]:
net.runtime_pollWait(0x7f6364cc1330, 0x72, 0xc8200101d0)
        /root/.gvm/gos/go1.5/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820122530, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820122530, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc8201224d0, 0x0, 0x7f63684dd828, 0xc82000f0a0)
        /root/.gvm/gos/go1.5/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc8201100b0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc8201100b0, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/tcpsock_posix.go:264 +0x3d
github.com/influxdb/influxdb/tcp.(*Mux).Serve(0xc82010d240, 0x7f6364cc0388, 0xc8201100b0, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tcp/mux.go:48 +0x4b
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open.func1
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:345 +0xae9

goroutine 20 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc8201100b8, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/meta.(*raftLayer).Accept(0xc820175a10, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/meta/store.go:2059 +0x51
github.com/hashicorp/raft.(*NetworkTransport).listen(0xc820122540)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/net_transport.go:346 +0x79
created by github.com/hashicorp/raft.NewNetworkTransport
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/net_transport.go:138 +0x284

goroutine 21 [select]:
github.com/hashicorp/raft.(*Raft).leaderLoop(0xc820192000)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/raft.go:849 +0xcce
github.com/hashicorp/raft.(*Raft).runLeader(0xc820192000)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/raft.go:821 +0x936
github.com/hashicorp/raft.(*Raft).run(0xc820192000)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/raft.go:602 +0xbb
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.run)-fm()
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/raft.go:252 +0x20
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc820192000, 0xc82017e8d0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/state.go:152 +0x44
created by github.com/hashicorp/raft.(*raftState).goFunc
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/state.go:153 +0x4d

goroutine 22 [select]:
github.com/hashicorp/raft.(*Raft).runFSM(0xc820192000)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/raft.go:509 +0x10eb
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.runFSM)-fm()
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/raft.go:253 +0x20
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc820192000, 0xc82017e8e0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/state.go:152 +0x44
created by github.com/hashicorp/raft.(*raftState).goFunc
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/state.go:153 +0x4d

goroutine 23 [select]:
github.com/hashicorp/raft.(*Raft).runSnapshots(0xc820192000)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/raft.go:1636 +0x493
github.com/hashicorp/raft.(*Raft).(github.com/hashicorp/raft.runSnapshots)-fm()
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/raft.go:254 +0x20
github.com/hashicorp/raft.(*raftState).goFunc.func1(0xc820192000, 0xc82017e8f0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/state.go:152 +0x44
created by github.com/hashicorp/raft.(*raftState).goFunc
        /tmp/tmp.d2Sk65XhgQ/src/github.com/hashicorp/raft/state.go:153 +0x4d

goroutine 24 [select]:
github.com/influxdb/influxdb/meta.(*localRaft).logLeaderChanges(0xc820106f00)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/meta/state.go:187 +0x4f5
created by github.com/influxdb/influxdb/meta.(*localRaft).open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/meta/state.go:177 +0x1344

goroutine 7 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc8201100c0, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/meta.(*Store).serveExecListener(0xc820126280)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/meta/store.go:622 +0xa2
created by github.com/influxdb/influxdb/meta.(*Store).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/meta/store.go:239 +0x364

goroutine 8 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc8201100c8, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/meta.(*Store).serveRPCListener(0xc820126280)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/meta/store.go:726 +0xa2
created by github.com/influxdb/influxdb/meta.(*Store).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/meta/store.go:242 +0x3b6

goroutine 26 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Server).monitorErrorChan(0xc82011f680, 0xc8201186c0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:491 +0x145
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Server).Open.func1
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cmd/influxd/run/server.go:351 +0xc67

goroutine 27 [select]:
github.com/influxdb/influxdb/monitor.(*Monitor).storeStatistics(0xc820116750)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/monitor/service.go:333 +0x187c
created by github.com/influxdb/influxdb/monitor.(*Monitor).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/monitor/service.go:134 +0x30c

goroutine 28 [select]:
github.com/influxdb/influxdb/services/hh.(*Service).retryWrites(0xc8201381b0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/hh/service.go:103 +0x314
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/hh/service.go:67 +0x210

goroutine 29 [select]:
github.com/influxdb/influxdb/services/hh.(*Service).expireWrites(0xc8201381b0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/hh/service.go:121 +0x2d1
created by github.com/influxdb/influxdb/services/hh.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/hh/service.go:68 +0x232

goroutine 30 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc8201100d0, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/cluster.(*Service).serve(0xc8201223f0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cluster/service.go:84 +0xdb
created by github.com/influxdb/influxdb/cluster.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cluster/service.go:61 +0x135

goroutine 31 [select]:
github.com/influxdb/influxdb/services/precreator.(*Service).runPrecreation(0xc82010ce80)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/precreator/service.go:75 +0x44f
created by github.com/influxdb/influxdb/services/precreator.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/precreator/service.go:53 +0x21f

goroutine 32 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc8201100d8, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/services/snapshotter.(*Service).serve(0xc8201068c0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/snapshotter/service.go:75 +0xa9
created by github.com/influxdb/influxdb/services/snapshotter.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/snapshotter/service.go:48 +0x130

goroutine 33 [chan receive]:
github.com/influxdb/influxdb/tcp.(*listener).Accept(0xc8201100e0, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tcp/mux.go:129 +0x6f
github.com/influxdb/influxdb/services/copier.(*Service).serve(0xc82010cf00)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/copier/service.go:78 +0xa9
created by github.com/influxdb/influxdb/services/copier.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/copier/service.go:51 +0x130

goroutine 34 [IO wait]:
net.runtime_pollWait(0x7f6364cc1270, 0x72, 0xc8200101d0)
        /root/.gvm/gos/go1.5/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc8201228b0, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc8201228b0, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc820122850, 0x0, 0x7f63684dd828, 0xc8201b0dc0)
        /root/.gvm/gos/go1.5/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820110160, 0x455470, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc820110160, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/tcpsock_posix.go:264 +0x3d
net/http.(*Server).Serve(0xc8201ae8a0, 0x7f6364cc0388, 0xc820110160, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/http/server.go:1887 +0xb3
net/http.Serve(0x7f6364cc0388, 0xc820110160, 0x7f636427a820, 0xc8201b4810, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/http/server.go:1765 +0x85
github.com/influxdb/influxdb/services/admin.(*Service).serve(0xc8201069b0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/admin/service.go:107 +0x12f
created by github.com/influxdb/influxdb/services/admin.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/admin/service.go:70 +0x491

goroutine 35 [select]:
github.com/influxdb/influxdb/services/continuous_querier.(*Service).backgroundLoop(0xc820138360)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:192 +0x565
created by github.com/influxdb/influxdb/services/continuous_querier.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/continuous_querier/service.go:126 +0x2a4

goroutine 36 [IO wait]:
net.runtime_pollWait(0x7f6364cc11b0, 0x72, 0xc8200101d0)
        /root/.gvm/gos/go1.5/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820122920, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820122920, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc8201228c0, 0x0, 0x7f63684dd828, 0xc820205460)
        /root/.gvm/gos/go1.5/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820110168, 0xc820038dc0, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc820110168, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/tcpsock_posix.go:264 +0x3d
net/http.(*Server).Serve(0xc820119860, 0x7f6364cc0388, 0xc820110168, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/http/server.go:1887 +0xb3
net/http.Serve(0x7f6364cc0388, 0xc820110168, 0x7f6364cc18f0, 0xc820115000, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/http/server.go:1765 +0x85
github.com/influxdb/influxdb/services/httpd.(*Service).serve(0xc820118b40)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/httpd/service.go:133 +0x6e
created by github.com/influxdb/influxdb/services/httpd.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/httpd/service.go:101 +0x6ca

goroutine 37 [select]:
github.com/influxdb/influxdb/tsdb.(*PointBatcher).Start.func2(0xc820119b00, 0xc8202040e0, 0xc820204100, 0xc820110188, 0xc820110180)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tsdb/batcher.go:72 +0x522
created by github.com/influxdb/influxdb/tsdb.(*PointBatcher).Start
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tsdb/batcher.go:107 +0x17e

goroutine 39 [chan receive]:
github.com/influxdb/influxdb/services/opentsdb.(*chanListener).Accept(0xc820204180, 0x0, 0x0, 0x0, 0x0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/opentsdb/handler.go:149 +0x83
net/http.(*Server).Serve(0xc820242000, 0x7f6364236028, 0xc820204180, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/http/server.go:1887 +0xb3
github.com/influxdb/influxdb/services/opentsdb.(*Service).serveHTTP(0xc820128460)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/opentsdb/service.go:340 +0x1ca
created by github.com/influxdb/influxdb/services/opentsdb.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/opentsdb/service.go:160 +0xefd

goroutine 40 [IO wait]:
net.runtime_pollWait(0x7f6364cc10f0, 0x72, 0xc8200101d0)
        /root/.gvm/gos/go1.5/src/runtime/netpoll.go:157 +0x60
net.(*pollDesc).Wait(0xc820122990, 0x72, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:73 +0x3a
net.(*pollDesc).WaitRead(0xc820122990, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/fd_poll_runtime.go:78 +0x36
net.(*netFD).accept(0xc820122930, 0x0, 0x7f63684dd828, 0xc820239e00)
        /root/.gvm/gos/go1.5/src/net/fd_unix.go:408 +0x27c
net.(*TCPListener).AcceptTCP(0xc820110190, 0xc82019fee0, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/tcpsock_posix.go:254 +0x4d
net.(*TCPListener).Accept(0xc820110190, 0x0, 0x0, 0x0, 0x0)
        /root/.gvm/gos/go1.5/src/net/tcpsock_posix.go:264 +0x3d
github.com/influxdb/influxdb/services/opentsdb.(*Service).serve(0xc820128460)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/opentsdb/service.go:198 +0xae
created by github.com/influxdb/influxdb/services/opentsdb.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/opentsdb/service.go:161 +0xf22

goroutine 41 [select]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShardGroups(0xc820106b90)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/retention/service.go:69 +0x351
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/retention/service.go:45 +0x1b4

goroutine 42 [select]:
github.com/influxdb/influxdb/services/retention.(*Service).deleteShards(0xc820106b90)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/retention/service.go:102 +0x7a0
created by github.com/influxdb/influxdb/services/retention.(*Service).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/services/retention/service.go:46 +0x1d9

goroutine 43 [select]:
github.com/influxdb/influxdb/cmd/influxd/run.(*Command).monitorServerErrors(0xc820114180)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:138 +0x2c5
created by github.com/influxdb/influxdb/cmd/influxd/run.(*Command).Run
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/cmd/influxd/run/command.go:120 +0xeeb

goroutine 44 [select, locked to thread]:
runtime.gopark(0xd172f8, 0xc8201ecf28, 0xbd3100, 0x6, 0x18, 0x2)
        /root/.gvm/gos/go1.5/src/runtime/proc.go:185 +0x163
runtime.selectgoImpl(0xc8201ecf28, 0x0, 0x18)
        /root/.gvm/gos/go1.5/src/runtime/select.go:392 +0xa64
runtime.selectgo(0xc8201ecf28)
        /root/.gvm/gos/go1.5/src/runtime/select.go:212 +0x12
runtime.ensureSigM.func1()
        /root/.gvm/gos/go1.5/src/runtime/signal1_unix.go:227 +0x353
runtime.goexit()
        /root/.gvm/gos/go1.5/src/runtime/asm_amd64.s:1696 +0x1

goroutine 12 [select]:
github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).autoflusher(0xc8201d41c0, 0xc8201aecc0)
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:604 +0x567
created by github.com/influxdb/influxdb/tsdb/engine/wal.(*Log).Open
        /tmp/tmp.d2Sk65XhgQ/src/github.com/influxdb/influxdb/tsdb/engine/wal/wal.go:240 +0x627
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,7 @@
 ### Bugfixes
 - [#3457](https://github.com/influxdb/influxdb/issues/3457): [0.9.3] cannot select field names with prefix + "." that match the measurement name
 - [#4111](https://github.com/influxdb/influxdb/pull/4111): Update pre-commit hook for go vet composites
+- [#4136](https://github.com/influxdb/influxdb/pull/4136): Return an error-on-write if target retention policy does not exist. Thanks for the report @ymettier
 
 ## v0.9.4 [2015-09-14]
 

--- a/cluster/points_writer.go
+++ b/cluster/points_writer.go
@@ -174,6 +174,9 @@ func (w *PointsWriter) MapShards(wp *WritePointsRequest) (*ShardMapping, error) 
 	if err != nil {
 		return nil, err
 	}
+	if rp == nil {
+		return nil, influxdb.ErrRetentionPolicyNotFound(wp.RetentionPolicy)
+	}
 
 	for _, p := range wp.Points {
 		timeRanges[p.Time().Truncate(rp.ShardGroupDuration)] = nil

--- a/errors.go
+++ b/errors.go
@@ -18,6 +18,10 @@ var (
 
 func ErrDatabaseNotFound(name string) error { return fmt.Errorf("database not found: %s", name) }
 
+func ErrRetentionPolicyNotFound(name string) error {
+	return fmt.Errorf("retention policy not found: %s", name)
+}
+
 func ErrMeasurementNotFound(name string) error { return fmt.Errorf("measurement not found: %s", name) }
 
 func Errorf(format string, a ...interface{}) (err error) {

--- a/meta/store_test.go
+++ b/meta/store_test.go
@@ -274,6 +274,47 @@ func TestStore_CreateRetentionPolicy(t *testing.T) {
 	}
 }
 
+// Ensure the store can create and get a retention policy on a database.
+func TestStore_CreateAndGetRetentionPolicy(t *testing.T) {
+	t.Parallel()
+	s := MustOpenStore()
+	defer s.Close()
+
+	// Create an additional nodes and database.
+	if _, err := s.CreateNode("hostX"); err != nil {
+		t.Fatal(err)
+	} else if _, err := s.CreateDatabase("db0"); err != nil {
+		t.Fatal(err)
+	}
+
+	// Create policy on database.
+	if _, err := s.CreateRetentionPolicy("db0", &meta.RetentionPolicyInfo{
+		Name:     "rp0",
+		ReplicaN: 2,
+		Duration: 48 * time.Hour,
+	}); err != nil {
+		t.Fatal(err)
+	}
+
+	// Get the policy on database.
+	if rpi, err := s.RetentionPolicy("db0", "rp0"); err != nil {
+		t.Fatal(err)
+	} else if !reflect.DeepEqual(rpi, &meta.RetentionPolicyInfo{
+		Name:               "rp0",
+		ReplicaN:           2,
+		Duration:           48 * time.Hour,
+		ShardGroupDuration: 24 * time.Hour,
+	}) {
+		t.Fatalf("unexpected policy: %#v", rpi)
+	}
+
+	// Get non-existent policies.
+	if _, err := s.RetentionPolicy("db0", "rp0"); err != nil {
+		t.Fatal(err)
+	}
+
+}
+
 // Ensure the store can delete a retention policy.
 func TestStore_DropRetentionPolicy(t *testing.T) {
 	t.Parallel()

